### DISCUSSION TITLE
About 90% of all themes on 'Find Themes' has 'Bad Link, cannot continue"

### DISCUSSION THREAD

**Discussion by Usalabs** [e107inc/e107#4811] [Q&A Themes]
It seems that when people create and upload new themes they seem to delete them after a while but the information still resides in the script, that searches for and downloads themes, I say around 90%, because I can only successfully download a few themes, the rest literally have bad links, either the links down exists anymore or the links are malformed, but in any case, the 'bad links' should be removed, so that people that download the themes, can see the ones that actually do work.

**Comment by PH5HP** [comment]
Hi,

I noticed the same. I wonder if this is related that on the THEMES download website of E107 just one THEME is allowed to download. For a second THEME you have to wait some time. Maybe this generated the bad links, because I noticed that on another day some THEMES can be downloaded which generate a bad link a day earlier.

Very annoying this.

Kind regards,
Henk - https://fileserver.ph5hp.nl (if my main computer is online)

**Comment by Moc** [comment]
@CaMer0n I am noticing odd behaviour as well on e107.org when downloading themes/plugins. It looks like there may be a flood protection setting running a little too actively. Can you take a look please?

**Comment by CaMer0n** [comment]
@Moc I'm having trouble reproducing the issue. Are you downloading via the Theme Manager or directly via the website?

**Reply by Moc** [reply]
@CaMer0n Did some more testing. There are two issues:

1)  Via Theme Manager > Find Themes (`/e107_admin/theme.php?mode=main&action=online`). I didn't try all themes of course, but just randomly clicking through the the pages and some (older themes) I do get the "bad link" message on quite a lot of themes. 

2) When downloading on e107.org I do get both "Woah, slow down cowboy! Please wait a few seconds before viewing another page." and "Woah, slow down cowboy! Please wait a few minutes before downloading another file." rather quickly. 

![image](https://github.com/e107inc/e107/assets/577978/e8bb8c6f-19dd-4271-b1c6-484d1f30efcc)

**Comment by tgtje** [comment]
Just info: download on **themes( site)** : no issues.. still as it was before : and yes the (about) 40 seconds delay between downloads is still active.: you will be prompted if it is to quick... ala Wow slow down...
'same as it was before'    (and adding : download accelerator in use?... forget it..)
